test/k8sT/Policies.go

```diff
@@ -304,68 +304,79 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
 
 		}, 500)
 
-		It("Same Policy in the different namespaces", func() {
-			namespace := "second"
-			policy := fmt.Sprintf("%s -n %s", l3Policy, namespace)
-			demoManifest := fmt.Sprintf("%s -n %s", demoPath, namespace)
-			kubectl.NamespaceCreate(namespace)
-			kubectl.Apply(demoManifest)
-			defer func() {
-				kubectl.Delete(demoManifest).ExpectSuccess()
-				kubectl.Delete(policy).ExpectSuccess()
-				kubectl.NamespaceDelete(namespace).ExpectSuccess()
-
-			}()
-
-			pods, err := kubectl.WaitforPods(
-				namespace,
-				"-l zgroup=testapp", 300)
-			Expect(pods).To(BeTrue(), "testapp pods are not ready after timeout")
-			Expect(err).To(BeNil(), "testapp pods are not ready after timeout")
-
-			By("Applying Policy in namespace")
-			eps := kubectl.CiliumEndpointPolicyVersion(ciliumPod)
-			_, err = kubectl.CiliumPolicyAction(
-				helpers.KubeSystemNamespace, policy, helpers.KubectlApply, 300)
-			Expect(err).Should(BeNil(), "L3 Policy cannot be applied in %q namespace", namespace)
-
-			err = helpers.WaitUntilEndpointUpdates(ciliumPod, eps, 4, kubectl)
-			Expect(err).Should(BeNil())
+		Context("Different namespaces", func() {
+
+			var (
+				namespace    = "second"
+				policy       = fmt.Sprintf("%s -n %s", l3Policy, namespace)
+				demoManifest = fmt.Sprintf("%s -n %s", demoPath, namespace)
+			)
+
+			BeforeEach(func() {
+				res := kubectl.NamespaceCreate(namespace)
+				res.ExpectSuccess("unable to create namespace %s: %s", namespace, res.CombineOutput())
+				res = kubectl.Apply(demoManifest)
+				res.ExpectSuccess("unable to apply manifest %s: %s", demoManifest, res.CombineOutput())
+			})
 
-			By("Applying Policy in default namespace")
-			eps = kubectl.CiliumEndpointPolicyVersion(ciliumPod)
-			_, err = kubectl.CiliumPolicyAction(
-				helpers.KubeSystemNamespace, l3Policy, helpers.KubectlApply, 300)
-			Expect(err).Should(BeNil(), "L3 Policy cannot be applied in %q namespace", helpers.DefaultNamespace)
-			err = helpers.WaitUntilEndpointUpdates(ciliumPod, eps, 4, kubectl)
-			Expect(err).Should(BeNil(), "Endpoints timeout on namespaces %q", helpers.DefaultNamespace)
+			AfterEach(func() {
+				// Explicitly do not check results to avoid incomplete teardown of test.
+				_ = kubectl.Delete(demoManifest).ExpectSuccess()
```

Comment:
I think that in the AfterEach we should not use ExpectSucess. It should continue always.

Reply:
Sorry, I missed this `ExpectSuccess`. The comment above I wrote even indicates that we don't want to check the result. Will fix this in next upload.